src/pybind/rados/setup.py

```diff
@@ -90,12 +88,20 @@ def check_sanity():
     compiler = new_compiler()
     customize_compiler(compiler)
 
+    if set(['MAKEFLAGS', 'MFLAGS', 'MAKELEVEL']).issubset(set(os.environ.keys())):
+        # The setup.py has been invoked by a top-level Ceph make.
+        # Set the appropriate CFLAGS and LDFLAGS
+        CEPH_ROOT = subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).strip().decode('utf-8')
+
+        compiler.add_include_dir(os.path.join(CEPH_ROOT, 'src', 'include'))
```

Comment:
`CFLAGS` and `LDFLAGS` are passed to `setup.py` explicitly in `Distutils.cmake` and `Makefile.am`, i think they would ready the `compiler` to do its job? iirc, distutils do check the env variables to figure out the cflags and ldflags.